spec/tbd_tests_spec.rb

```diff
@@ -4869,7 +5155,267 @@
+    expect(right_fine_wall.setConstruction(construction2)).to be(true)
+
+    subs = front_office_wall.subSurfaces
+    expect(subs.empty?).to be(false)
+    expect(subs.size).to eq(4)
+
+    psi_set = "poor (BETBG)"
+    ioP = File.dirname(__FILE__) + "/../json/tbd_warehouse9.json"
+    # {
+    #   "schema": "https://github.com/rd2/tbd/blob/master/tbd.schema.json",
+    #   "description": "testing error detection",
+    #   "psis": [
+    #     {
+    #       "id": "detailed_2",
+    #       "fenestration": 0.600
+    #     },
+    #     {
+    #       "id": "regular (BETBG)",                    <<<< non-fatal ERROR # 1
+    #       "fenestration": 0.700
+    #     }
+    #   ],
+    #   "khis": [
+    #     {
+    #       "id": "cantilevered beam",
+    #       "point": 0.6
+    #     }
+    #   ],
+    #   "surfaces": [
+    #     {
+    #       "id": "Office Front Wall",
+    #       "khis": [
+    #         {
+    #           "id": "beam",                           <<<< non-fatal ERROR # 2
+    #           "count": 3
+    #         }
+    #       ]
+    #     },
+    #     {
+    #       "id": "Office Left Wall",
+    #       "khis": [
+    #         {
+    #           "id": "cantilevered beam",
+    #           "count": 300                           <<<< triggers WARNING # 1
+    #         }
+    #       ]
+    #     }
+    #   ],
+    #   "edges": [
+    #     {
+    #       "psi": "detailed",                          <<<< non-fatal ERROR # 3
+    #       "type": "fenestration",
+    #       "surfaces": [
+    #         "Office Front Wall",
+    #         "Office Front Wall Window 1"
+    #       ]
+    #     }
+    #   ]
+    # }
+    schemaP = File.dirname(__FILE__) + "/../tbd.schema.json"
+    io, surfaces = processTBD(os_model, psi_set, ioP, schemaP)
+    expect(io.nil?).to be(false)
+    expect(io.is_a?(Hash)).to be(true)
+    expect(io.empty?).to be(false)
+    expect(io.has_key?(:edges))
+    expect(surfaces.nil?).to be(false)
+    expect(surfaces.is_a?(Hash)).to be(true)
+    expect(surfaces.size).to eq(23)
+    expect(surfaces.has_key?("Office Front Wall")).to be(true)
+    expect(surfaces["Office Front Wall"].has_key?(:edges)).to be(true)
+    expect(surfaces.has_key?("Office Left Wall")).to be(true)
+    expect(surfaces["Office Left Wall"].has_key?(:edges)).to be(true)
+    expect(surfaces.has_key?("Fine Storage Right Wall")).to be(true)
+    expect(surfaces["Fine Storage Right Wall"].has_key?(:edges)).to be(true)
+
+    expect(TBD.status).to eq(TBD::ERROR)
+    expect(TBD.logs.size).to eq(6)
+    # TBD.logs.each { |log| puts log[:msg] }
+    #   'clerestory' vertex count (3 or 4)
+    #   Can't override 'regular (BETBG)' PSI set  - skipping
+    #   'Office Front Wall' KHI 'beam' mismatch
+    #   'Office Front Wall' edge PSI set mismatch - skipping
+    #   Can't assign 180.007 W/K to 'Office Left Wall' - too conductive
+    #   Can't derate 'Fine Storage Right Wall' - material already derated
```

Comment:
In normal circumstances, and as noted earlier, these log messages will make their way to both "tbd.log" & "tbd.out.json" files (enriched with their recorded timestamps and respective log levels e.g., "ERROR"). Thinking of changing the JSON schema to conserve _key:value_ pairs for :time, :log_level &. :messages (might as well make it easier for 3rd-party applications to parse logged information). @macumber : thoughts?

Reply:
Schema is now enriched to log a single timestamp (everything happens under ~1 sec), log objects are kept distinct (facilitates subsequent parsing), etc.

Reply:
I agree that a timestamp for the overall conversion is just as valuable as the timestamp for each log message (as long as the messages are kept in order so you can tell which errors happen first).